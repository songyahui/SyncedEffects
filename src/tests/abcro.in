A;
B;
C;
;
A B;
A B;
R C;
A B;
A B;
R;
C B;
A;
A;
B;
R B;
A;
;
B;
R;
C;
A B;
