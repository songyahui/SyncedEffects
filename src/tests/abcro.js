"use hopscript"

var hh = require("hiphop");

var inSig = {direction: hh.IN};
var outSig = {direction: hh.OUT};

var prg =
    <hh.module A=${inSig} B=${inSig} C=${inSig} R=${inSig} O=${outSig}>
      <hh.loopeach R>
	<hh.parallel>
	  <hh.await A/>
	  <hh.await B/>
	  <hh.await C/>
	</hh.parallel>
	<hh.emit O/>
      </hh.loopeach>
    </hh.module>;

exports.prg = new hh.ReactiveMachine(prg, "ABCRO");
