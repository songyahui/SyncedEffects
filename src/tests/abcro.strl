module ABCRO :

input A, B, C, R;
output O;

loop
  [
    await A
  ||
    await B
  ||
    await C
  ];
  emit O
each R

end module