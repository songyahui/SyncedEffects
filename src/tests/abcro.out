ABCRO> A;
--- Output:
ABCRO> B;
--- Output:
ABCRO> C;
--- Output:
ABCRO> ;
--- Output:
ABCRO> A B;
--- Output: O
ABCRO> A B;
--- Output:
ABCRO> R C;
--- Output:
ABCRO> A B;
--- Output:
ABCRO> A B;
--- Output:
ABCRO> R;
--- Output:
ABCRO> C B;
--- Output:
ABCRO> A;
--- Output: O
ABCRO> A;
--- Output:
ABCRO> B;
--- Output:
ABCRO> R B;
--- Output:
ABCRO> A;
--- Output:
ABCRO> ;
--- Output:
ABCRO> B;
--- Output:
ABCRO> R;
--- Output:
ABCRO> C;
--- Output:
ABCRO> A B;
--- Output: O
